apps/dashboard/app/models/files.rb

```diff
@@ -36,6 +48,48 @@ def self.stat(path)
     }
   end
 
+  def self.can_download_as_zip?(path, timeout: Configuration.file_download_dir_timeout, max_download_as_zip_size: Configuration.file_download_dir_max)
+    path = Pathname.new(path)
+
+    can_download = false
+    error = nil
+
+    if ! (path.directory? && path.readable? && path.executable?)
+      error = I18n.t('dashboard.files_directory_download_unauthorized')
+    else
+      # Determine the size of the directory.
+      o, e, s = Open3.capture3("timeout", "#{timeout}s", "du", "-cbs", path.to_s)
+
+      # Catch SIGTERM.
+      if s.exitstatus == 124
+        error = I18n.t('dashboard.files_directory_size_calculation_timeout')
+      elsif ! s.success?
+        error = I18n.t('dashboard.files_directory_size_unknown', exit_code: s, error: e)
+      else
+        # Example output from: du -cbs $path
+        # 
+        #    496184  .
+        #    64      ./ood-portal-generator/lib/ood_portal_generator
+        #    72      ./ood-portal-generator/lib
+        #    24      ./ood-portal-generator/templates
+        #    40      ./ood-portal-generator/share
+        #    576     ./ood-portal-generator
+        #
+        size = o&.split&.first
+
+        if size.blank?
+          error = I18n.t('dashboard.files_directory_size_parse_error')
+        elsif size.to_i > max_download_as_zip_size
+          error = I18n.t('dashboard.files_directory_too_large', max_download_as_zip_size: max_download_as_zip_size)
```

Comment:
`max_download_as_zip_size` is admiin facing (since they may change the string) and should be `download_directory_size_limit` or something similarly agnostic as to the package/compression choice